plugins/bil_pipe/bil_pipe_main.ml

```diff
@@ -0,0 +1,62 @@
+open Core_kernel
+open Bap.Std
+open Bil_const_propagation
+open Bil_renum_virtual
+open Bil_memo
+include Self ()
+
+let parse = function
+  | "enable" -> `Ok true
+  | "disable" -> `Ok false
+  | s ->
+    `Error (sprintf "unknown value %s, possible are: disable | enable" s)
+
+let print fmt s =
+  Format.fprintf fmt "%s" @@
+  match s with
+  | true  -> "enable"
+  | false -> "disable"
+
+let switcher ?(default=true) =
+  Config.(param (converter parse print default) ~default)
+
+let simpl = Bil.fold_consts
```

Comment:
so, simplification is actually the what you call propagate consts and is a fixed point of `FC . PC . DE`. 

I want to see the following options:

```
--bil-normalization=<level> -- defaults to 1 (with 0 keeps it denormalized)
--bil-optimization=<level> -- defaults to 1 (or 0, I don't know yet)
```

I'm not sure that we need to provide right now options between whether dead code elimination should be applied, or constant folding. I think we can start with the simple choice: optimization=0 - no simplifications are applied. optimization=1 -- all optimizations are applied.